src/rules/onfido-idv.js

```diff
@@ -0,0 +1,108 @@
+ *    - ONFIDO_REGION: The supported Onfido region your tenant is operating in
+ *
+ * @param {object} user
+ * @param {object} context
+ * @param {function} callback
+ */
+async function redirectOnfidoRule(user, context, callback) {
+  // using auth0 rule-utilities to make sure our rule is efficient in the pipeline
+  const { Auth0RedirectRuleUtilities } = require('@auth0/rule-utilities@0.1.0');
+  // requiring Onfido's node SDK for making the calls easier to Onfido's service. 
+  const { Onfido, Region } = require('@onfido/api@1.5.1');
+
+  const ruleUtils = new Auth0RedirectRuleUtilities(user, context, configuration);
+
+  // creating a claim namespace for adding the Onfido IDV check results back to the ID Token
+  const claimNamespace = 'https://claims.onfido.com/';
+
+  // creating a new Onfido client, the region here is where your Onfido instance is located. Possible values are EU for Europe, US for United States, and CA for Canada.
+  const onfidoClient = new Onfido({
+    apiToken: configuration.ONFIDO_API_TOKEN,
+    region:
+      configuration.ONFIDO_REGION === 'EU' ? Region.EU : configuration.ONFIDO_REGION === 'US' ? Region.US : configuration.ONFIDO_REGION === 'CA' ? Region.CA : Region.EU,
+  });
+
+  user.app_metadata = user.app_metadata || {};
+
+  if (ruleUtils.isRedirectCallback && ruleUtils.queryParams.session_token) {
+    // User is back from the Onfido experience and has a session token to validate and assign to user meta
+
+    // Validating session token and extracting payload for check results
+    let payload;
+    try {
+      payload = ruleUtils.validateSessionToken();
+    } catch (error) {
+      return callback(error);
+    }
+
+    // assigning check status and result to the app_metadata so the downstream application can decided what to do next
+    // note, in the example integration, the Onfido app returns after 30 seconds even if the check is still in progress 
+    // If this claim status is still in_progress it is recommended the downstream application recheck for completion or implement the Onfido Webhook: https://documentation.onfido.com/#webhooks
+    // Additionally, you can place these items into the idToken claim with custom claims as needed as shown
+    const onfido = {
+      check_result: payload.checkResult,
+      check_status: payload.checkStatus,
+      applicant_id: payload.applicant,
+    };
+    try {
+      await auth0.users.updateAppMetadata(user.user_id, onfido);
+    } catch (error) {
+      callback(error);
+    }
+
+    user.app_metadata.onfido = idv;
+
+    context.idToken[claimNamespace + 'check_result'] = payload.checkResult;
+    context.idToken[claimNamespace + 'check_status'] = payload.checkStatus;
+    context.idToken[claimNamespace + 'applicant_id'] = payload.applicant;
+
+    return callback(null, user, context);
+  }
+
+  if (ruleUtils.canRedirect && (user.app_metadata.onfido === undefined || user.app_metadata.onfido.check_status === '')) {
```

Comment:
```suggestion
  if (ruleUtils.canRedirect && !user.app_metadata.onfido.check_status) {
```
See suggestion above.

Reply:
Are there any other statuses that might prompt a re-check?

Reply:
No. Basically because it is async, it would be up to the downstream to determine what to do next.